tools/GenerateCompilationDatabase.ps1

```
@@ -0,0 +1,166 @@
+Param(
+    [string]$Ms2ccVersion = "1.3.0",
+    [switch]$Update
+
+)
+
+$binlogFileBase = Join-Path (Split-Path $PSScriptRoot -parent) "BuildOutput\Binlogs"
+$binlogFiles = Get-ChildItem -Path $binlogFileBase -Filter *.binlog -Recurse | Select-Object -ExpandProperty FullName
+
+Write-Host "Binlog files found:"
+$binlogFiles | ForEach-Object { Write-Host $_ }
+
+$VCToolsInstallDir = . "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -prerelease -requires Microsoft.Component.MSBuild -property InstallationPath
+write-host "VCToolsInstallDir: $VCToolsInstallDir"
+
+$msBuildPath = "$VCToolsInstallDir\MSBuild\Current\Bin\msbuild.exe"
+write-host "msBuildPath: $msBuildPath"
+
+Remove-Item "temp-filtered.log" -ErrorAction SilentlyContinue
```

A few things:
1. "temp-filtered.log" will be removed from the current directory. Better to use an explicit temp location
  * 2nd best answer = `$filteredLog = Join-Path ${env:Temp} "GenerateCompilationDatabase-filtered.log"`
  * Best answer =
```
$root = Get-ProjectRoot
$temp = Join-Path $root 'temp'
$filteredLog = Join-Path $temp 'GenerateCompilationDatabase-filtered.log'
```
2. `-Force` option to help ensure the file can be removed
3. Why SilentlyContinue? If the file can't be removed won't it cause problems later when trying to write (or worse, append) to it?
   3.1. If you really want to best-effort the file deletion you may want `Ignore`. Like `SilentlyContinue` but doesn't record the error in `$Error`. But I suspect better to `-Force` and let it fail if the file can't be deleted
4. If you need to specify an error action the short form's usually preferred e.g. `... -EA:SilentlyContinue`
4. Remove-Item can emit output. Unless that's really desired you probably want `$null = Remove-Item...`